easybuild/tools/config.py

```diff
@@ -590,9 +590,29 @@ def update_build_option(key, value):
     """
     # BuildOptions() is a (singleton) frozen dict, so this is less straightforward that it seems...
     build_options = BuildOptions()
+    orig_value = build_options._FrozenDict__dict[key]
     build_options._FrozenDict__dict[key] = value
     _log.warning("Build option '%s' was updated to: %s", key, build_option(key))
 
+    # Return original value, so they can be restored later if needed
+    return orig_value
+
+
+def update_build_options(key_value_dict):
+    """
+    Update all build options in the key_value_dict with the value given in that dictionary,
+    by calling update_build_option(key, value) repeatedly.
+    This function can be used e.g. when EasyConfig-specific build options are passed in an EasyStack file.
+    See https://github.com/easybuilders/easybuild-framework/issues/3513#issuecomment-986990195
```

Comment:
In which contexts this can be used is irrelevant for this particular imho, and I wouldn't mention variable or even function names, that's just more places to adjust if we happen to do some refactoring.

```suggestion
Update build options as specified by the given dictionary (where keys are assumed to be build option names).
Returns dictionary with original values for the updated build options.
```